### PR TITLE
doc/stdenv: how to mkDerivation with no srcs

### PR DESCRIPTION
**Pull request by chris-martin**
Copying the lesson learned from #23099 into the manual.

### CODE REVIEW

**Comment by vcunat** [comment]
I think this rather belongs into the general part describing phases, as one may want to disable others as well.  [There is hint](http://nixos.org/nixpkgs/manual/#sec-stdenv-phases) that you can override whole phases, so I imagine we might add that it's often useful to disable individual phases by `namePhase = "true";` and add a note that `preName` and `postName` are also skipped in that case (or if you override whole phase).  What do you think?

File: doc/stdenv.xml [hunk 1]
```diff
@@ -457,7 +457,10 @@ Additional file types can be supported by setting the
   <varlistentry>
     <term><varname>srcs</varname> / <varname>src</varname></term>
     <listitem><para>The list of source files or directories to be
-    unpacked or copied.  One of these must be set.</para></listitem>
+    unpacked or copied.  One of these must be set. To make a
+    derivation with no sources, you can avoid this requirement by
+    setting <literal>unpackPhase = "true";</literal> to replace the
+    default unpack phase with a noop.</para></listitem>
```
Comment (copumpkin) [inline doc/stdenv.xml]: Or just use `runCommand`?

Comment (vcunat) [inline doc/stdenv.xml]: Yes, that's the usual approach if you want to get rid of *all* phases.

**Comment by chris-martin** [comment]
However we fix it, the statement

> One of these must be set.

must be qualified in some way, because right now it's flatly untrue.

**Comment by chris-martin** [comment]
I suppose we could also add a hint to the error message?

**Comment by vcunat** [comment]
Anything related to a phase is void if you disable that phase &ndash; I'd think that doesn't need mentioning explicitly.

**Comment by chris-martin** [comment]
I think user-friendly documentation explicitly addresses particular problems that people are likely to run into, in the specific place where they're likely to look for the solution - rather than requiring everyone to read the entire manual and having a complete understanding of how everything comes together.

**Comment by chris-martin** [comment]
@copumpkin Can you elaborate on `runCommand`? The manual doesn't mention it at all.

**Comment by 7c6f434c** [comment]
There is `runCommand` function that takes three arguments (name, attributes to pass to `mkDerivation` —  most will become environment variables by default, and the command to run). It doesn't do anything by default, so it is used when you want just to concatenate two files into the output file or something like that.

**Comment by chris-martin** [comment]
It sounds like the consensus is that `runCommand` is the way to go here, so withdrawing this and putting in issue #25507 instead.

**Comment by Mic92** [comment]
mkDerivation without unpack phase is still useful because of its unpack phase: https://github.com/nixos-users/wiki/wiki/General-recommendations#package-simple-custom-scripts

**Comment by chris-martin** [comment]
I don't know, I've lost all energy on this PR, it seems like an uphill battle.

**Pull request closed without merging**